From ab560ddcb1cf95e246d7fa785bf1155aebeadd87 Mon Sep 17 00:00:00 2001
From: Pavel Dubrova <pashadubrova@gmail.com>
Date: Tue, 16 Nov 2021 18:09:33 +0200
Subject: [PATCH 088/175] power: qpnp-fg-gen3: Use module_platform_driver

The qpnp-fg-gen3 driver init and exit functions are calling
platform_driver_register and platform_driver_unregister and
doing nothing else. This same as that of the module_platform_driver,
remove this init and exit functions and use the module_platform_driver
instead.

Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>
---
 drivers/power/supply/qcom/qpnp-fg-gen3.c | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/drivers/power/supply/qcom/qpnp-fg-gen3.c b/drivers/power/supply/qcom/qpnp-fg-gen3.c
index f23f36d1b749..e255a0abdbca 100644
--- a/drivers/power/supply/qcom/qpnp-fg-gen3.c
+++ b/drivers/power/supply/qcom/qpnp-fg-gen3.c
@@ -5727,18 +5727,7 @@ static struct platform_driver fg_gen3_driver = {
 	.shutdown	= fg_gen3_shutdown,
 };
 
-static int __init fg_gen3_init(void)
-{
-	return platform_driver_register(&fg_gen3_driver);
-}
-
-static void __exit fg_gen3_exit(void)
-{
-	return platform_driver_unregister(&fg_gen3_driver);
-}
-
-module_init(fg_gen3_init);
-module_exit(fg_gen3_exit);
+module_platform_driver(fg_gen3_driver);
 
 MODULE_DESCRIPTION("QPNP Fuel gauge GEN3 driver");
 MODULE_LICENSE("GPL v2");
-- 
2.48.1

